pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -274,19 +274,27 @@ def __init__(
         self,
         coeffs,
         observables,
+        reorder_fn=None,
         pulses: List["HardwarePulse"] = None,
         settings: Union["RydbergSettings", "TransmonSettings"] = None,
     ):
         self.settings = settings
         self.pulses = [] if pulses is None else pulses
+
+        def trivial_func(_, y):
+            return y
+
+        self.reorder_fn = trivial_func if reorder_fn is None else reorder_fn
```

Comment:
I'm not sure if this trivial function is necessary. `HardwareHamiltonian` is not exposed to the user, so any case where a reodering function needs to be specified, we can assume that it will be, so we can just have it be `None` is one isn't given.

Reply:
I actually reverted to just making the default the `_reorder_parameters` function since it acts trivially whenever there is no `AmplitudeAndPhase` callable objects in the params